server/src/main/java/com/hedera/block/server/verification/hasher/CommonUtils.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.verification.hasher;
+
+import com.hedera.hapi.block.BlockItemUnparsed;
+import com.hedera.hapi.block.stream.BlockProof;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.crypto.DigestType;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+/**
+ * Provides common utility methods for hashing and combining hashes.
+ */
+public final class CommonUtils {
+    private CommonUtils() {
+        throw new UnsupportedOperationException("Utility Class");
+    }
+
+    /**
+     * The size of an SHA-384 hash in bytes.
+     */
+    public static final int HASH_SIZE = DigestType.SHA_384.digestLength();
+
+    /**
+     * The tag for the SHA-384 algorithm.
+     */
+    private static final String sha384HashTag = "SHA-384";
+
+    /**
+     * Returns the SHA-384 hash of the given bytes.
+     * @param bytes the bytes to hash
+     * @return the SHA-384 hash of the given bytes
+     */
+    public static Bytes noThrowSha384HashOf(final Bytes bytes) {
```

Comment:
could add the non-null annotation to the parameter for the static code analysis

Reply:
✅ Done!